h2/src/main/org/h2/command/Parser.java

```diff
@@ -4861,7 +4861,7 @@ private String readIdentifierWithSchema2(String s) {
         schemaName = s;
         s = readColumnIdentifier();
         if (currentTokenType == DOT) {
-            if (equalsToken(schemaName, database.getShortName())) {
+            if (equalsToken(schemaName, database.getShortName()) || database.getIgnoreCatalogs()) {
```

Comment:
If a catalog name is equal to a schema name, what will happen?

Reply:
Nothing special, the condition still be true. That's OK, but sides of condition could be swapped. 

The whole condition shouldn't be used by normal applications because H2 does not support catalogs, so it's efficiency is not very critical.